### PR TITLE
Updates for new interfaces

### PR DESCRIPTION
**Pull request by NerdDiffer**
Hi, I was interested in contributing to this API client. 

It's been a few months since Backpack.tf added some interfaces and upgraded a version of one of them, so I made some updates. I also made a few changes to the way query string parameters are set. Please review and let me know what you think.

Thanks,
Rafael

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -15,20 +18,20 @@ function backpacktf(apiKey, app) {
     throw new TypeError('Expected `apiKey` to be a String');
   }
 
-  this.apiKey = apiKey;
-  this.app = app || 440;
+  this.basicOptions = {
+    key: apiKey,
+    appid: app || 440,
```
Comment (KenanY) [inline index.js]: Trailing comma.

**Comment by KenanY** [comment]
Am I wrong in saying that the use of `lodash.assign` would mutate `this.basicOptions` and potentially affect future function calls?

**Comment by KenanY** [comment]
Also I prefer single quotes to double quotes. I also think we can do with moving `getPriceHistoryOptions` to `index.js` instead of putting it in a new file.

This contribution is great though! Looking forward to merging :)

File: test/helpers.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  t.deepEqual(helpers.getPriceHistoryOptions({
+    item: 'myItemName',
+    quality: 'Haunted'
+  }), {
+    item: 'myItemName',
+    quality: 'Haunted',
+    tradable: '1',
+    craftable: '1',
+    priceindex: '0',
```
Comment (KenanY) [inline test/helpers.js]: Trailing comma.

**Comment by NerdDiffer** [comment]
Thanks for the feedback. You are right, calling `lodash.assign` on the `basicOptions` property will mutate it. I wrote some tests to confirm this.

So, I'm a little confused. Did you need me to make some of these changes? Just in case, I have something pushed to my fork.

It seems kind of important to test the setting of query string parameters (or...at least, what goes into it), but without mutating the original value of `basicOptions`. I don't think it needs public exposure though. So, I thought keeping the helper module might be a good idea, if only to make testing easier.

**Comment by NerdDiffer** [comment]
I should add, I'm not sure how to design it so these, effectively private, methods are testable. I'm pretty sure that they're important enough to be tested, but I may be wrong. But also, the two methods in the `helper` module don't need to be directly called by the client. Therefore, they are not exported as part of the backpacktf object. 

Is keeping those 'private-ish' methods in a separate module the best way to do this?

**Comment by kenany** [comment]
@NerdDiffer Amended, squashed, and landed as https://github.com/KenanY/backpack.tf/commit/f2b0e22d757c27677b7e70556679b02436285ff9. Thank you so much for this contribution! I changed many little-things during my amending process but you still retain all authorship. In a better world, I should have merged this as several commits:
- update `IGetUserListings` endpoint
- add `IGetMarketPrices`
- add `IGetPriceHistory`
- fix `getPrices()` test
- etc.

but once I had the monolithic commit I relented to just merging that :)

I have also added you as a collaborator of this repository. But let's please continue using pull requests for the sake of reviewing each other's code :)

**Comment by NerdDiffer** [comment]
@KenanY, Sounds good. Thanks! I'll use pull requests in the future, and I like your use of the `assign` module to handle the `basicOptions` property.

For future reference, would it have been easier if I had spread out the proposed changes among more commits?

**Comment by kenany** [comment]
@NerdDiffer Yes, more commits/pull requests would result in smaller diffs to review :)

**Pull request closed without merging**